tools/pylib/boututils/check_scaling.py

```diff
@@ -0,0 +1,70 @@
+"""Functions for checking the error scaling of MMS or MES results
+
+"""
+
+from numpy import array, isclose, log, polyfit
+
+
+def get_order(grid_spacing, errors):
+    """Get the convergence order of errors over the full range of
+    grid_spacing, and at small spacings
+
+    Parameters
+    ----------
+    grid_spacing : list of float
+        The grid spacing or inverse of number of grid points
+    errors : list of float
+        The error at each grid spacing
+
+    Returns
+    -------
+    tuple of float
+        The first value is the error scaling over the full range of
+        grid spacings; the second value is the scaling over the last
+        two points
+
+    """
+    full_range = polyfit(log(grid_spacing), log(errors), 1)
+
+    small_spacing = log(errors[-2] / errors[-1]) / log(grid_spacing[-2] / grid_spacing[-1])
+
+    return (full_range[0], small_spacing)
+
+
+def check_order(actual_order, expected_order, tolerance=2.e-1):
+    """Check if the actual_order is sufficiently close to the
+    expected_order within a given tolerance
+
+    """
+    return isclose(actual_order, expected_order, rtol=tolerance)
```

Comment:
I would not have expected that the tolerance increases with the order.
Do we intend it to be this way?
`rtol=0, atol=tolerance` is what I would have expected.
[From the documentation](https://docs.scipy.org/doc/numpy/reference/generated/numpy.isclose.html)
```
    absolute(a - b) <= (atol + rtol * absolute(b))
```

Reply:
Again, we're not very consistent with this. In some places we use an absolute tolerance and in others a relative tolerance. I don't think it makes a great deal of difference either way, to be honest.

Reply:
Sorry, I meant practically it doesn't seem to make any difference. I'm not sure why you would expect it to behave one way or the other anyway?